pkg/client/dtclient/settings_client.go

```diff
@@ -580,7 +581,14 @@ func (d *SettingsClient) Upsert(ctx context.Context, obj SettingsObject, upsertO
 		retrySetting = *upsertOptions.OverrideRetry
 	}
 
-	resp, err := SendWithRetryWithInitialTry(ctx, d.client.POST, d.settingsObjectAPIPath, corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests}, payload, retrySetting)
+	resp, err := coreapi.AsResponseOrError(
```

Comment:
Please use `NewResponseFromHTTPResponse` instead of the deprecated `AsResponseOrError`